includes/class-wc-gateway-amazon-payments-advanced-abstract.php

```diff
@@ -949,7 +949,8 @@ public function has_v1_settings() {
 	 * @return bool
 	 */
 	public function has_other_gateways_enabled() {
-		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
```

Comment:
Shouldn't we check that we don't have this function available? I believe in frontend we should use the old way, since it won't be instantiated twice.

If that's not available (in admin) we should fall back.

Reply:
The problem here wasn't the availability of the function, the problem is that it removes the gateway from the WC instance if the settings are empty, thing that happens in any new install.

Reply:
The function is available, but it remove the gateway from the WC instance if you check it there, if the gateway is not enabled/settings are empty, this way the settings or the gateway itself disappear on a clean install.

Reply:
The function is available, but it removes the gateway from the WC instance if you check it there, if the gateway is not enabled, this way the settings and the gateway itself disappear on a clean install.